uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -37,198 +39,127 @@ define([
     var initialLeaveRequestAttributes = {}; // used to compare the change in leaverequest in edit mode
     var listeners = [];
     var loggedInContactId = '';
-    var role = '';
-    var vm = this;
     var NO_ENTITLEMENT_ERROR = 'No entitlement';
+    var role = '';
+    var vm = _.assign(this, directiveOptions); // put all directive options directly in the vm
```

Comment:
The directive options are actually props for the component/controller itself. Normally it is supposed to be inside the vm directly as we do with other components. But in this case those were enclosed in the `directiveOptions`, because other wise all the values needs to be injected individually.